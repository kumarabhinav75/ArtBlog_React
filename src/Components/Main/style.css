.AllCards {
  width: 75%;
  margin: 0 auto;
  display: flex;
  justify-content: space-between;
  flex-wrap: wrap;
}

.createPostButton {
  display: flex;
  justify-content: center;
  margin : 2%;
}