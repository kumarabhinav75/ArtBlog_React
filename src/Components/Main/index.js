import React, {Component} from 'react';
import Header from '../Header';
import Footer from '../Footer';
import AllCards from '../AllCards';
import Button from '../Button';
// import cardJSON from '../../mockData/index.json';
import './style.css';

const axios = require('axios');

class Main extends Component {
  state = {
    cardJSON: [],
  }

  componentDidMount() {
    axios.get('https://api.myjson.com/bins/hc5ye')
      .then((jsonObj) => {
        console.log(jsonObj);
        this.setState({cardJSON: jsonObj.data});
      });
  }

  render(){
    return (
      <div>
      <Header />
      <div className="AllCards">
        <AllCards cardsJSON = {this.state.cardJSON}/>
      </div>
      <div className="createPostButton">
        <Button onClick= {() => (window.location.href = "/posts")} />
      </div>
      <Footer />
    </div>
    );
  }
}


export default Main;