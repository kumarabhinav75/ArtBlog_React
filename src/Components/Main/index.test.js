import React from 'react';
import renderer from 'react-test-renderer';
import { shallow } from 'enzyme';
import Main from './index';

describe(' Main ', () => {
  it ('should match the snapshot', () => {
    const tree = renderer.create(<Main />);
    expect(tree).toMatchSnapshot();
  });

  it('Should call componentDidMount once', async () => {
    const wrapper = shallow(<Main />);
    const spy = jest.spyOn(wrapper.instance(),'componentDidMount');
    const didMount = wrapper.instance().componentDidMount();
    expect(spy).toHaveBeenCalled();
    expect(wrapper.instance().state.cardJSON).toEqual(['a']);
  });
})