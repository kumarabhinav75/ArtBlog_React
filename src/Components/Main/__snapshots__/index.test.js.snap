// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[` Main  should match the snapshot 1`] = `
<div>
  <header>
    <div
      className="container"
    >
      <p
        className="heading"
      >
        The Artifact
      </p>
      <p
        className="heading-content"
      >
        <i>
          Culture & Art blog
        </i>
      </p>
      <ul
        className="nav-list"
      >
        <div
          className="nav-items"
        >
          <li>
            Blog
          </li>
          <li>
            About
          </li>
          <li>
            Contact
          </li>
        </div>
      </ul>
    </div>
  </header>
  <div
    className="AllCards"
  />
  <div
    className="createPostButton"
  >
    <button
      className="create-button"
      onClick={[Function]}
      type="button"
    >
      Create Post
    </button>
  </div>
  <footer>
    <div
      className="container-width"
    >
      <p>
        Â© artifact.com 2019
      </p>
    </div>
  </footer>
</div>
`;
